platform-sdk/docs/proposals/consensus-layer/Consensus-Layer.md

```diff
@@ -0,0 +1,720 @@
+
+For example, in the above diagram, each event has exactly 2 parents: the self-parent, and one other-parent. Event `B3`
+has as a self-parent `B2`, and an other-parent of `A2`. Events `A1`, `A2`, `A3`, and `A4` are all events created by node
+`A`, while `B1`, `B2`, `B3`, and `B4` are created by node `B`, and so on.
+
+Each node has its own copy of the hashgraph. Since events are being gossiped asynchronously throughout the network,
+newer events (those at the top of the graph) may be known to some nodes, and not to others. Broken or dishonest nodes
+may work to prevent some events from being known to all nodes, and thus there may be some differences in the hashgraph
+of each node. But the hashgraph algorithm will, with probability 1, deterministically come to consensus given just over
+2/3 of the nodes are honest.
+
+#### The Road to Finality
+
+A transaction is submitted to a node in the network. This node, upon verifying the integrity of the transaction, will
+include this transaction in the next event it creates. This new event is assigned a _birth round_ matching the most
+recent round number of the hashgraph on the node that created the event. This birth round lets other nodes in the
+network know how far along in processing the hashgraph this node was at the time the event was created.
+
+The event is then gossiped, or distributed throughout the network. Each node that receives this event validates it and
+inserts it into their own copy of the hashgraph. Eventually, the hashgraph algorithm runs on each node (which may
+happen at different wall clock times!) and the event is included in a round. Every honest node will always include the
+event in the same round.
+
+Each round is then passed to the Execution layer, where the transactions in the round are executed, and the state is
+transitioned accordingly. For example, hbars may be transferred from one account to another. At the end of some
+deterministic number of rounds, the Execution layer will create a block. The block hash will be signed and all the
+nodes together will work to sign the block (using an algorithm known as TSS). The block is then exported from the
+node.
+
+Once the block is exported from the node, the transaction execution is truly final. Since the network together signed
+the block, users have an iron-clad guarantee that the contents of the block represent the consensus result of executing
+those transactions. Since they are included in a blockchain, there is an immutable, cryptographic proof of execution.
+
+### Liveness Under Stress
+
+A node is under stress when it is unable to process events fast enough to keep up with the other nodes in the network.
+The network is under stress when a sufficient number of nodes are under stress. When the network is under stress, all
+nodes together must work to reduce the number of transactions allowed into the network, to give the network a chance
```

Comment:
This is correct, but I think I'll skip adding the clarification, since it's stated that this mechanism will be more precisely defined in a future proposal.